server/pulp/server/content/web/views.py

```diff
@@ -144,4 +156,13 @@ def get(self, request):
             return self.redirect(request, self.key)
 
         # NotFound
-        return HttpResponseNotFound(request.path_info)
+        raise Http404
+
+    @staticmethod
+    def directory_index(path):
+        listing = os.listdir(path)
+        context = {
+            'dirs': [f for f in listing if os.path.isdir(os.path.join(path, f))],
+            'files': [f for f in listing if os.path.isfile(os.path.join(path, f))]
```

Comment:
It also made me sad and I  briefly wondered if there was a better way (that was also readable).